geodata/models/raster/reader.py

```diff
@@ -0,0 +1,128 @@
+        elif len(raster_querry) == 1:
+            self.raster_entry = raster_querry.first()
+            # Clear out associated entries because they could be invalid
+            BandMetaEntry.objects.filter(parent_raster=self.raster_entry).delete()
+            ConvertedRasterFile.objects.filter(source_raster=self.raster_entry).delete()
+        else:
+            # This should never happen because it is a foreign key
+            raise RuntimeError('multiple raster entries found for this file.')
+
+        self.raster_entry.raster_file = self.rfe
+
+        with rasterio.open(file_path) as src:
+            self.raster_entry.number_of_bands = src.count
+            self.raster_entry.driver = src.driver
+            self.raster_entry.crs = src.crs.to_proj4()
+            self.raster_entry.origin = [src.bounds.left, src.bounds.top]
+            self.raster_entry.extent = [
+                src.bounds.left,
+                src.bounds.bottom,
+                src.bounds.right,
+                src.bounds.top,
+            ]
+            self.raster_entry.resolution = src.res
+            self.raster_entry.height = src.shape[0]
+            self.raster_entry.width = src.shape[1]
+            self.raster_entry.transform = src.transform.to_gdal()  # TODO: check this
+            # A catch-all metadata feild:
+            # TODO: self.raster_entry.metadata =
+
+            # thumbnail = RasterEntryReader.create_thumbnail(src)
+            # self.raster_entry.thumbnail.save('thumbnail.jpg', thumbnail, save=True)
+
+            coords = (
+                (src.bounds.left, src.bounds.top),
+                (src.bounds.right, src.bounds.top),
+                (src.bounds.right, src.bounds.bottom),
+                (src.bounds.left, src.bounds.bottom),
+                (src.bounds.left, src.bounds.top),  # Close the loop
+            )
+
+            spatial_ref = SpatialReference(src.crs.to_wkt())
+            logger.info(f'Raster footprint SRID: {spatial_ref.srid}')
+            # This will convert the Polygon to the DB's SRID
+            self.raster_entry.footprint = Polygon(coords, srid=spatial_ref.srid)
+
+            # TODO: now create a `BandMetaEntry` entry for each band
+            #       this will collect statistics on the bands
+            self.band_entries = []
+            for i in range(src.count):
+                band_meta = BandMetaEntry()
+                band_meta.parent_raster = self.raster_entry
+                try:
+                    band_meta.description = src.descriptions[i]
+                except IndexError:
+                    pass
+                try:
+                    band_meta.nodata_value = src.nodatavals[i]
+                except IndexError:
+                    pass
+                try:
+                    band_meta.dtype = src.dtypes[i]
+                except IndexError:
+                    pass
+                # TODO: Compute statistics: min, max, mean, std
```

Comment:
Many files store min, max, mean, std for each band.  These don't always match the data.

If we add code to compute these, it might be useful to actually generate a histogram of values, too (though then we have to decide on binning, etc.).

Reply:
For now, I will update this PR to use the values stored directly in the raster. Any further implementation warrants a new issue and subsequent PR in my opinion as we'll need t put a bit of thought into how to compute these stats and bins if the values in the raster are not valid.